src/modm/architecture/interface/spi_master.hpp

```diff
@@ -156,6 +129,40 @@ class SpiMaster : public ::modm::PeripheralDriver, public Spi
 	static modm::ResumableResult<void>
 	transfer(const uint8_t *tx, uint8_t *rx, std::size_t length);
 #endif
+
+public:
+	/**
+	 * Request access to the spi master within a context.
+	 * You may acquire the spi master multiple times within the same context.
+	 *
+	 * The configuration handler will only be called when aquiring the spi
+	 * master for the first time (if it is not a `nullptr`).
+	 *
+	 * @warning		Aquires must be balanced with releases of the **same** context!
+	 * @warning		Aquires are persistent even after calling `initialize()`!
+	 *
+	 * @return	`0` if another context is using the spi master, otherwise
+	 * 			`>0` as the number of times this context acquired the master.
+	 */
+	static uint8_t
+	acquire(void *ctx, ConfigurationHandler handler = nullptr);
+
+	/**
+	 * Release access to the spi master within a context.
+	 *
+	 * @warning		Releases must be balanced with acquires of the **same** context!
+	 * @warning		Releases are persistent even after calling `initialize()`!
+	 *
+	 * @return	`0` if nothing can be released anymore (for any context)
+	 * 			`>0` as the number of times this context can still release the master.
+	 */
+	static uint8_t
+	release(void *ctx);
+
+private:
+	static inline uint8_t count{0};
+	static inline void* context{nullptr};
+	static inline ConfigurationHandler configuration{nullptr};
```

Comment:
So… what happens when you use two SPI busses?

Reply:
Nothing good, but it's fixable. We can move the acquire/release stuff to a separate class and make it a CRTP-like template. Then we have separate variables for each class using it.

```cpp
template <typename Spi>
class SpiLock
{
 	static inline uint8_t count{0};
…
public:
    static uint8_t
    acquire(void *ctx, ConfigurationHandler handler = nullptr);
};
```
and then use it as:
```cpp
class SpiMaster1 : public SpiLock<SpiMaster1>, …
{
…
};
```

Reply:
Ok great, cos this is a great cleanup!

Reply:
@salkinium fixed